sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -39,11 +39,12 @@ object DefaultOptimizer extends Optimizer {
     Batch("Distinct", FixedPoint(100),
       ReplaceDistinctWithAggregate) ::
     Batch("Operator Optimizations", FixedPoint(100),
-      UnionPushdown,
+      UnionPushDown,
       CombineFilters,
       PushPredicateThroughProject,
       PushPredicateThroughGenerate,
       ColumnPruning,
+      LimitPushDown,
```

Comment:
Actually I'm think of adding a test for the whole batch. As this batch is getting larger and larger, there are a lot more interactions between rules, we should make sure there is no conflict between rules and rules are all at right order.
cc @marmbrus @rxin